docs/howto/operator/gcp/dataproc.rst

```diff
@@ -0,0 +1,188 @@
+file system. You can specify a file:/// path to refer to a local file on a cluster's master node.
+
+The job configuration can be submitted by using:
+:class:`~airflow.providers.google.cloud.operators.dataproc.DataprocSubmitJobOperator`.
+
+.. exampleinclude:: ../../../../airflow/providers/google/cloud/example_dags/example_dataproc.py
+    :language: python
+    :dedent: 4
+    :start-after: [START how_to_cloud_dataproc_submit_job_to_cluster_operator]
+    :end-before: [END how_to_cloud_dataproc_submit_job_to_cluster_operator]
+
+Examples of job configurations to submit
+----------------------------------------
+
+We have provided an example for every framework below.
+There are more arguments to provide in the jobs than the examples show. For the complete list of arguments take a look at
+`DataProc Job arguments <https://cloud.google.com/dataproc/docs/reference/rest/v1/projects.regions.jobs>`__
+
+Example of the configuration for a PySpark Job:
+
+.. exampleinclude:: ../../../../airflow/providers/google/cloud/example_dags/example_dataproc.py
+    :language: python
+    :dedent: 4
+    :start-after: [START how_to_cloud_dataproc_pyspark_config]
+    :end-before: [END how_to_cloud_dataproc_pyspark_config]
+
+Example of the configuration for a SparkSQl Job:
+
+.. exampleinclude:: ../../../../airflow/providers/google/cloud/example_dags/example_dataproc.py
+    :language: python
+    :dedent: 4
+    :start-after: [START how_to_cloud_dataproc_sparksql_config]
+    :end-before: [END how_to_cloud_dataproc_sparksql_config]
+
+Example of the configuration for a Spark Job:
+
+.. exampleinclude:: ../../../../airflow/providers/google/cloud/example_dags/example_dataproc.py
+    :language: python
+    :dedent: 4
+    :start-after: [START how_to_cloud_dataproc_spark_config]
+    :end-before: [END how_to_cloud_dataproc_spark_config]
+
+Example of the configuration for a Hive Job:
+
+.. exampleinclude:: ../../../../airflow/providers/google/cloud/example_dags/example_dataproc.py
+    :language: python
+    :dedent: 4
+    :start-after: [START how_to_cloud_dataproc_hive_config]
+    :end-before: [END how_to_cloud_dataproc_hive_config]
+
+Example of the configuration for a Hadoop Job:
+
+.. exampleinclude:: ../../../../airflow/providers/google/cloud/example_dags/example_dataproc.py
+    :language: python
+    :dedent: 4
+    :start-after: [START how_to_cloud_dataproc_hadoop_config]
+    :end-before: [END how_to_cloud_dataproc_hadoop_config]
+
+Example of the configuration for a Pig Job:
+
+.. exampleinclude:: ../../../../airflow/providers/google/cloud/example_dags/example_dataproc.py
+    :language: python
+    :dedent: 4
+    :start-after: [START how_to_cloud_dataproc_pig_config]
+    :end-before: [END how_to_cloud_dataproc_pig_config]
+
+
+Example of the configuration for a SparkR:
+
+.. exampleinclude:: ../../../../airflow/providers/google/cloud/example_dags/example_dataproc.py
+    :language: python
+    :dedent: 4
```

Comment:
```suggestion
    :dedent: 0
```

Reply:
My bad, copied it straight from the other guides and was not aware of its function. Will adjust it

Reply:
@joppevos we don't have to set all dedents to 0. Only where it is necessary.

For example:
```python
# [START anhor]
zero_indent_dict = {"a": 1}
# [END anhor]

# [START anhor2]
with DAG():
    op = MyOperator()
# [END anhor2]
```
then in docs:
```rst
.. exampleinclude:: ../../../../airflow/providers/google/cloud/example_dags/example_dataproc.py
    :language: python
    :dedent: 0
    :start-after: [START anhor]
    :end-before: [END anhor]
```
because the first anhor has no indent in linked python file. But the second code snipped is indented by 4 spaces so to make code look better in docs we do:

```rst
.. exampleinclude:: ../../../../airflow/providers/google/cloud/example_dags/example_dataproc.py
    :language: python
    :dedent: 4
    :start-after: [START anhor2]
    :end-before: [END anhor2]
```
in this way it will render to:
```
op = MyOperator()
```
not to
```
    op = MyOperator()
```

Reply:
@turbaszek Thanks for the explanations! clears it up